server/src/main/java/org/opensearch/cluster/metadata/WeightedRoundRobinRoutingMetadata.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.metadata;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.Version;
+import org.opensearch.cluster.AbstractNamedDiffable;
+import org.opensearch.cluster.NamedDiff;
+import org.opensearch.cluster.routing.WRRWeights;
+import org.opensearch.common.Strings;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Contains metadata for weighted round-robin shard routing weights
+ *
+ * @opensearch.internal
+ */
+public class WeightedRoundRobinRoutingMetadata extends AbstractNamedDiffable<Metadata.Custom> implements Metadata.Custom {
```

Comment:
Here and elsewhere should we avoid `WeightedRoundRobinRouting` and lets call `OperationRouting`  or `ShardRouting` instead

Reply:
We are specifically using this to store weighted round robin shard routing weights metadata. Giving it a generic name can't be confusing?

Reply:
Can we atleast omit `RoundRobin` and `RR` from the files name?

Reply:
Should we  call it WeightedShardRoutingMetadata?

Reply:
I am fine with either of these `WeightedRoutingMetadata` or `WeightedOperationRoutingMetadata`

Reply:
Made changes